src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -1,69 +1,76 @@
-import React, { useCallback, useEffect, useState } from "react";
+import React, { useCallback, useEffect, useState, ChangeEvent } from "react";
+import classNames from "classnames";
+
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
 
 import { VenueEvent } from "types/venues";
 import { Room, RoomTypes } from "types/rooms";
 import { User } from "types/User";
 
 import { WithId } from "utils/id";
-import { currentVenueSelectorData, venueEventsSelector } from "utils/selectors";
 import { isTruthy } from "utils/types";
+import { formatToWeekdayFullname, formatUtcSeconds } from "utils/time";
+import { currentVenueSelectorData, venueEventsSelector } from "utils/selectors";
 
 import { useWorldUsers } from "hooks/users";
 import { useSelector } from "hooks/useSelector";
 import { useProfileModalControls } from "hooks/useProfileModalControls";
 
+import { InputField } from "components/atoms/InputField";
 import { RoomModal } from "components/templates/PartyMap/components";
 
+import { NavSearchResult } from "./NavSearchResult";
+
 import "./NavSearchBar.scss";
-import { InputField } from "components/atoms/InputField";
-import { faSearch } from "@fortawesome/free-solid-svg-icons";
 
-interface SearchResult {
-  rooms: Room[];
-  users: readonly WithId<User>[];
-  events: VenueEvent[];
-}
+type WithImageUrl<T extends object> = T & { image_url?: string };
```

Comment:
What's this used for? Should it be moved to our utility types?

Reply:
hmm, I don't think that it's needed in the utilities.
It's used for Event search results. Events originally don't have icons and I extend its type, adding `image_url` property from the room it is related to.

do you think I should move the type to utilities? and what do you think about snake_case here? I used same property name as the `Room` has, but maybe I should use snakeCase instead as this is another interface?

Reply:
`lowerCamelCase` is definitely our preferred method, so we should stick to that. The few places that use `snake_case` are legacy/because that's what firebase passed us and we never changed it.

If you only envisage it being needed here for this specific use case then it's probably fine to leave here. If you think it might be beneficial to reuse in future, then i'd say move it into the utility types.

Reply:
removed the type in the commit: https://github.com/sparkletown/sparkle/pull/1328/commits/c50dba7f0cdf5c19441f06c1edd9889d40e3cb16
as I moved `JSX.Element[]` into `useMemo<T>`.

Reply:
Sounds good :)